target/core/kvtrans/dss_kvtrans_module.c

```diff
@@ -818,11 +818,19 @@ void dss_setup_kvtrans_req(dss_request_t *req, dss_key_t *k, dss_value_t *v)
     switch(req->opc) {
         case DSS_NVMF_KV_IO_OPC_STORE:
             kreq->req.opc = KVTRANS_OPC_STORE;
-            DSS_ASSERT(v->value);
+            if (v->length != 0) {
+                // This check is in place to support
+                // writing zero byte keys
+                DSS_ASSERT(v->value);
```

Comment:
Why not return error code, why to assert ?

Reply:
Firstly, I agree to not have asserts in customer code. Here are the reasons, why I think it is OK to have assert here.

This API has a void declared for its return. 

Maybe just return, instead of asserting; however this assert gives us an opportunity to understand or catch some issues. This is a non-trivial check.

This assert is only triggered when the RDMA stack is sending some value buffer with some length but the data ptr is empty. This should never happen; however if someone is capable of doing this to RDMA stack, this gives us an opportunity to trap that behavior instantly.

This assert if logged, should terminate target as there is something really wrong.

Let me know your thoughts.